### PR TITLE
Send beacon for unique visitors in JspmTest and JspmControl

### PR DESCRIPTION
**Pull request by OliverJAsh**
We want to know if the ~1.5% drop in unique visitors and 3% drop in page views for JspmTest is because of an error in SystemJS or something beforehand (i.e. inaccurate bucketing or bounce due to bigger payload). We can use the data from these beacons and compare it with the data in Omniture.

/cc @gklopper @sndrs

### CODE REVIEW

File: common/app/templates/headerInlineJS/cloudwatchBeacons.scala.js [hunk 1]
```diff
@@ -83,5 +83,21 @@
...
         }
+
+        // Send beacon for unique visitors in JspmTest and JspmControl server-side test variants
+        if (guardian.config.switches.jspmTestUniqueVisitorsBeacon) {
+            var tests = guardian.config.tests;
+            var inTest = tests.jspmTest || tests.jspmControl;
+            var localStorageKey = 'gu.jspm-test.visited';
+            // Protect browsers with no localStorage
+            try {
+                var visited = !!window.localStorage.getItem(localStorageKey)
```
Comment (gklopper) [inline common/app/templates/headerInlineJS/cloudwatchBeacons.scala.js]: Is this wrapped in `isModernBrowser`?

File: common/app/templates/headerInlineJS/cloudwatchBeacons.scala.js [hunk 2]
```diff
@@ -83,5 +83,21 @@
...
+                if (!visited && inTest) {
+                    var beaconName = tests.jspmTest ? 'jspm-test' : 'jspm-control';
```
Comment (gklopper) [inline common/app/templates/headerInlineJS/cloudwatchBeacons.scala.js]: You sure its not just `var beaconName = tests.jspmTest;`?

File: common/app/templates/headerInlineJS/cloudwatchBeacons.scala.js (hunk 1 above)
Comment (OliverJAsh) [inline common/app/templates/headerInlineJS/cloudwatchBeacons.scala.js]: See e14fb1dce1d33e85135a65f6d2b45530a0332736

**Comment by gklopper** [comment]
:+1:

File: common/app/templates/headerInlineJS/cloudwatchBeacons.scala.js [hunk 3]
```diff
@@ -83,5 +83,19 @@
...
             }
         }
+
+        // Send beacon for unique visitors in JspmTest and JspmControl server-side test variants
+        // Requires localStorage, so modern browsers only
+        if (window.guardian.isModernBrowser && guardian.config.switches.jspmTestUniqueVisitorsBeacon) {
+            var tests = guardian.config.tests;
+            var inTest = tests.jspmTest || tests.jspmControl;
+            var localStorageKey = 'gu.jspm-test.visited';
+            var visited = !!window.localStorage.getItem(localStorageKey)
```
Comment (sndrs) [inline common/app/templates/headerInlineJS/cloudwatchBeacons.scala.js]: `localStorage` needs to be in a `try/catch`

**Comment by sndrs** [comment]
:+1: apart from comment

**Comment by gklopper** [comment]
:+1:

**Pull request merged**